velox/core/Expressions.h

```diff
@@ -61,7 +61,12 @@ class ConstantTypedExpr : public ITypedExpr {
   // Variant::null() value is supported.
   ConstantTypedExpr(TypePtr type, Variant value)
       : ITypedExpr{ExprKind::kConstant, std::move(type)},
-        value_{std::move(value)} {}
+        value_{std::move(value)} {
+    VELOX_CHECK_EQ(
+        value_.kind(),
```

Comment:
This check is not sufficient for complex types. Use isTypeCompatible instead. Please, add tests to ensure that ConstantTypedExpr(ARRAY(VARCHAR()), variant::array({1,2,3}) fails.

Reply:
Thanks @mbasmanova, updated to use `isTypeCompatible`. Could you PTAL?